R/PipeOpTrafo.R

```diff
@@ -277,111 +290,108 @@ mlr_pipeops$add("targetinverter", PipeOpTargetInverter)
-#' g$add_edge(src_id = "regr.rpart", dst_id = "targetinverter",
+#' g$add_edge(src_id = "regr.rpart", dst_id = "targetinvert",
 #'   src_channel = 1, dst_channel = 2)
 #'
 #' g$train(task)
 #' g$predict(task)
 #'
 #' #syntactic sugar using ppl():
 #' tt = ppl("targettrafo", graph = PipeOpLearner$new(LearnerRegrRpart$new()))
-#' tt$param_set$values$targettrafosimple.trafo = function(x) log(x, base = 2)
-#' tt$param_set$values$targettrafosimple.inverter = function(x) 2 ^ x
+#' tt$param_set$values$targetmutate.trafo = function(x) log(x, base = 2)
+#' tt$param_set$values$targetmutate.inverter = function(x) list(response = 2 ^ x$response)
 #' @family PipeOps
 #' @include PipeOp.R
 #' @export
-PipeOpTargetTrafoSimple = R6Class("PipeOpTargetTrafoSimple",
+PipeOpTargetMutate = R6Class("PipeOpTargetMutate",
   inherit = PipeOpTargetTrafo,
   public = list(
-    initialize = function(id = "targettrafosimple", param_vals = list()) {
-      ps = ParamSet$new(params = list(
-        ParamUty$new("trafo", default = identity, tags = c("train", "predict"), custom_check = function(x) check_function(x, nargs = 1L)),
-        ParamUty$new("inverter", default = identity, tags = "predict", custom_check = function(x) check_function(x, nargs = 1L)),
-        ParamUty$new("new_target_name", tags = c("train", "predict"), custom_check = function(x) check_character(x, any.missing = FALSE, len = 1L)),
-        ParamUty$new("new_task_type", tags = c("train", "predict"), custom_check = function(x) check_choice(x, choices = mlr_reflections$task_types$type))
-        )
-      )
-      # FIXME: add condition for new_target_name and new_task_type on trafo and inverter when mlr-org/paradox#278 has an answer
-      ps$values = list(trafo = identity, inverter = identity)
-      super$initialize(id = id, param_set = ps, param_vals = param_vals)
-    },
-
-    set_state = function(task) {
-      list()
-    },
+    initialize = function(id = "targetmutate", param_vals = list(), new_task_type = NULL) {
+      private$.new_task_type = assert_choice(new_task_type, mlr_reflections$task_types$type, null.ok = TRUE)
+      ps = ParamSet$new(list(
+        ParamUty$new("trafo", tags = c("train", "predict"), custom_check = function(x) check_function(x, nargs = 1L)),
+        ParamUty$new("inverter", tags = "predict", custom_check = function(x) check_function(x, nargs = 1L))
+      ))
+      # We could add a condition here for new_task_type on trafo and inverter when mlr-org/paradox#278 has an answer.
+      # HOWEVER conditions are broken in paradox, it is a terrible idea to use them in PipeOps,
+      # see https://github.com/mlr-org/paradox/issues/216 and related comment in PipeOpLearnerCV
 
-    train_target = function(task) {
-      # handle trafo = identity separately
-      if (identical(self$param_set$values$trafo, identity)) {
-        return(task)  # early exit
+      ps$values = list(trafo = identity, inverter = function(x) list(response = identity(x$response)))
```

Comment:
why does `identity` not work? because `x` still contains the `$truth` column? In that case `invert` should probably remove the `truth` col and make `identity` work here.

Reply:
because `x` also contains the `row_id` as a column. So I guess I will simply drop this as well